sceptre/template.py

```diff
@@ -291,8 +290,50 @@ def _render_jinja_template(template_dir, filename, jinja_vars):
         logger.debug("%s Rendering CloudFormation template", filename)
         env = jinja2.Environment(
             loader=jinja2.FileSystemLoader(template_dir),
-            undefined=jinja2.StrictUndefined
+            undefined=jinja2.StrictUndefined,
         )
+        env.filters.update(self.jinja_filters)
         template = env.get_template(filename)
         body = template.render(**jinja_vars)
         return body
+
+    @property
+    def jinja_filters(self):
+        """
+        Returns cached jinja filters if already computed, or loads/caches
+        the filters from python files if this is the first time.
+
+        This (optional) feature relies on the environment variable
+        ${SCEPTRE_JINJA_FILTER_ROOT} because we don't have easy access
+        from here to CLI parsing info or global configuration info.
+        """
+        # give back cached jinja filters if available
+        if hasattr(self, '_jinja_filters'):
+            return self._jinja_filters
+        # load jinja filters if not already cached
+        env_var_name = 'SCEPTRE_JINJA_FILTER_ROOT'
```

Comment:
I don't think we should be reaching env_vars directly, can we read this as environment config for consistency

Reply:
How exactly should that be done?  `Template()` seem to only be instantiated with with scepre_user_data, not environment_config.  See also this issue https://github.com/cloudreach/sceptre/issues/355

Does a path from environment_config or stack_config actually help here?  I think what is really needed is the exact value passed from the CLI argument `--dir`, because even given an environment or stack path, if env folders are nested then one couldn't be sure how far to go back up the directory tree to get to the sceptre root.